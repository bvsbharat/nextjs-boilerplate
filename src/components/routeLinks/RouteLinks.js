import React from "react";
import PropTypes from "prop-types";
import Link from "next/link";

const RouteLinks = ({ routes }) => {
    const renderRouteLinks = () => {
        if (!routes.length) {
            return false;
        }

        return routes.map((item, i) => (
            <li
                key={i}
                className="cursor-pointer h-full flex items-center text-sm text-gry-800 mx-10 tracking-normal"
            >
                <Link href={`/${item.link}`}>{item.name}</Link>
            </li>
        ));
    };

    return <>{renderRouteLinks()}</>;
};

RouteLinks.propTypes = {
    routes: PropTypes.array,
};

RouteLinks.defaultProps = {
    routes: [],
};

export default RouteLinks;
