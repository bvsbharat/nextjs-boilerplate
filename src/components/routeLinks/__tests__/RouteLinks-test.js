import React from "react";
import renderer from "react-test-renderer";
import RouteLinks from "../RouteLinks";

describe(RouteLinks, () => {
    it("Should render correctly", () => {
        const component = renderer.create(<RouteLinks />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
