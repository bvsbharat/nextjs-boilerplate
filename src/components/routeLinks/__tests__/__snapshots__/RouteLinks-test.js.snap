// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteLinks Should render correctly 1`] = `null`;
